### ISSUE TITLE
Nuclear Winter texture preview looks wrong

### ISSUE THREAD

**Issue by sprEEEzy** [Jazcash/spring-map-parser#29]
Colors wrong (green, white), mexes also invisible.

![image](https://github.com/user-attachments/assets/4818a3bb-3a8e-47b2-bb02-c887e3727f31)

Chobby map preview looks ok.

![image](https://github.com/user-attachments/assets/832736d1-1a24-4a2f-b45b-af1f724ee2c1)

**Comment by p2004a** [comment]
Chobby preview is just in game screenshot uploaded by mapper themselves, so it makes sense it looks ok.

**Comment by burnhamrobertp** [comment]
This appears to have been fixed at some point? Generating the texture.jpg presently (even on master, without the texture fix that's in PR right now) looks like this:

<img width="1473" height="741" alt="Image" src="https://github.com/user-attachments/assets/0956eb66-f062-467b-996b-db20e5cd8ada" />

**Comment by p2004a** [comment]
At all mipmap levels?

**Comment by burnhamrobertp** [comment]
No, I only texted at 16 on both master and the pull request branch with the color swap bit math fix. I'll retest other mipmap sizes and get back to you.

**Comment by burnhamrobertp** [comment]
| mipmapSize | Texture |
| --- | --- |
| 4 | <img width="1280" height="640" alt="Image" src="https://github.com/user-attachments/assets/bc11cd66-a2ad-4126-8131-c702978aaf14" /> |
| 8 | <img width="2560" height="1280" alt="Image" src="https://github.com/user-attachments/assets/1e33bbeb-33d9-4149-a40e-5209821f488c" /> |
| 16 | <img width="2847" height="1425" alt="Image" src="https://github.com/user-attachments/assets/5534e3a0-a818-4fdd-997b-387358850b07" /> (screenshot of file) |
| 32 | <img width="2846" height="1425" alt="Image" src="https://github.com/user-attachments/assets/1b4178b3-424c-488d-a79d-8bdb096a8e7c" /> (screenshot of file) |

The only other thing that occurs to me presently is the possibility that there is some part of the processing that occurs in maps-metadata that is introducing the artifacts? That occurs as part of the github action, though, right? Its not something I could run through and test locally?

**Comment by p2004a** [comment]
maps-metadata is not involved in generation of that preview, I believe that comes from https://bar-rts.com/maps/ which does it's own map parsing in https://github.com/beyond-all-reason/bar-db

It predates maps-metadata and so on.

**Comment by burnhamrobertp** [comment]
I was looking at https://www.beyondallreason.info/map/nuclear-winter does it also use the map-parsing in bar-db? I had thought much of the data on beyondallreason.info was the result of the maps-metadata, but perhaps I was mistaken? Or perhaps the data does come from maps-metadata, but the actual map is driven off of output from bar-db or another repo?

**Comment by p2004a** [comment]
Maps information at beyondallreason.info comes to webflow from maps-metadata. 

bar-rts.com is a separate site, backend is the linked bar-db repo, that does map parsing separately

**Comment by p2004a** [comment]
Ah, but the texture is also wrong at beyondallreason.info.

Here is the raw image as generated by the map-parser library invoked from maps-metadata parsing service: https://storage.googleapis.com/maps-cache-8512/Nuclear%20Winter%20Bar%201.4/cache-v2/texture.jpg

**Comment by burnhamrobertp** [comment]
So the fact that the preview for nuclear-winter on bar.info (linked above in my last comment) shows the same defects as the original issue reported, despite the fact that the "last updated" for that map on bar.info is quite recent (January) seems to suggest that the issue is in map-parser (which seems unlikely at this point) or in maps-metadata?

That is, unless its possible that the map images weren't replaced when the maps-metadata last updated this map in late January?

**Comment by p2004a** [comment]
Given that two independent places, maps-metadata and bar-db, that share only the fact that they use map-parser, have the same artifact, points that issue is in map-parser.

**Comment by burnhamrobertp** [comment]
With the output at all mipmapSizes looking correct; if you have any insight as to where you think the problem might lie, I'm happy to investigate it

**Comment by p2004a** [comment]
Looks like you made some error when testing @burnhamrobertp.

The export is working fine with #34 merged, but I get the exact some artifact before it. So it was fixed by your PR.

**Comment by burnhamrobertp** [comment]
Odd. I guess that makes sense but I'm not sure how I arrived at the earlier result. Perhaps a test without a build in between, though I had that happen a couple times early on and was pretty careful about it from them on.

Oh well, thanks for double checking Mr faulty conclusions
